scene/gui/check_box.cpp

```diff
@@ -75,14 +66,30 @@ Size2 CheckBox::get_minimum_size() const {
 	Size2 tex_size = get_icon_size();
 	minsize.width += tex_size.width;
 	if (get_text().length() > 0) {
-		minsize.width += MAX(0, get_theme_constant(SNAME("h_separation")));
+		minsize.width += MAX(0, theme_cache.h_separation);
 	}
-	Ref<StyleBox> sb = get_theme_stylebox(SNAME("normal"));
-	minsize.height = MAX(minsize.height, tex_size.height + sb->get_margin(SIDE_TOP) + sb->get_margin(SIDE_BOTTOM));
+	minsize.height = MAX(minsize.height, tex_size.height + theme_cache.normal_style->get_margin(SIDE_TOP) + theme_cache.normal_style->get_margin(SIDE_BOTTOM));
 
 	return minsize;
 }
 
+void CheckBox::_update_theme_item_cache() {
+	Button::_update_theme_item_cache();
```

Comment:
Does this super class call actually work?

IINM it will update `Button`'s `theme_cache` which is shadowed by `CheckBox`'s `theme_cache`, which is why you're having to re-query `h_separation`. `Button` and `CheckBox` have different and incompatible `ThemeCache` structs so you wouldn't be able to query cached items from `Button` until you redo it manually I think?

Reply:
Yep, learnt the hard way when working on the Window version of this PR. Forgot to add the super call and got issues. Basically, each class may have its own drawing or sizing logic encapsulated which needs its own private cache.

I'd love to be able to define the cache in a more dynamic, shared way, but I don't have ideas at the moment. This can be improved later, and the super calls can be eliminated by using GDVIRTUAL, I think.